src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt

```diff
@@ -5,13 +5,55 @@ import org.gradle.api.Incubating
 import org.gradle.api.Project
 import org.gradle.api.publish.maven.MavenPom
 import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.plugins.signing.SigningPlugin
 
 @Incubating
 @Suppress("UnnecessaryAbstractClass")
 abstract class MavenPublishBaseExtension(
   private val project: Project
 ) {
 
+  /**
+   * Automatically apply Gradle's `signing` plugin and configure all publications to be signed. If signing credentials
+   * are not configured this will fail the build unless the current version is a `SNAPSHOT`.
+   *
+   * Signing can be done using a local `secring.gpg` by setting these Gradle properties:
+   * ```
+   * signing.keyId=24875D73
+   * signing.password=secret
+   * signing.secretKeyRingFile=/Users/me/.gnupg/secring.gpg
+   * ```
+   *
+   * Alternatively an in memory key can be used by exporting an ascii-armored GPG key and setting these Gradle properties"
+   * ```
+   * signingInMemoryKey=exported_ascii_armored_key
```

Comment:
Maybe add a comment how to produce an armored key ? (`gpg --armour --export-secret-keys KEY_ID` or `gpg --armor --export-secret-keys KEY_ID` depending your part of the world :))

Also, do gradle properties handle new lines? If there's some kind of escaping required, it would be nice to give a more complete exemple:

```
# not sure that works?
signingInMemoryKey=-----BEGIN PGP PRIVATE KEY BLOCK-----
[...]
-----END PGP PRIVATE KEY BLOCK-----
```

Or if they don't handle newlines it might be ok to strip them all before adding the key to `gradle.properties`?

Reply:
You can just remove the line breaks to make it single line. You also need to leave out the first line and the last 2 lines.

For example if this is your private key export (using https://github.com/vanniktech/gradle-maven-publish-plugin/blob/master/src/integrationTest/fixtures/common/test-secring.gpg, password `test`)
```
-----BEGIN PGP PRIVATE KEY BLOCK-----

lQdGBF4jUfwBEACblZV4uBViHcYLOb2280tEpr64iB9b6YRkWil3EODiiLd9JS3V
7+BWpZVF8mbGy8AUR7T3GUsYmEvhHGw2s0IosOUVIu5W3eiU/K4CFMEsmV5JZLRS
...
Yeb/PX6yl+4vG17kFDiBkCPl+88lcbS5iHc1mXuWePlW+Q/6tN55Oy6mX8J/dGLV
BYBQKbmChGuRlZdwWJgHh+O5rN5aDAEda/aYqfgdUC8BHSwhCQPtyCIsOKjOYZzM
=JfHr
-----END PGP PRIVATE KEY BLOCK-----
```
You'd end change it to `lQdGBF4jUfwBEACblZV4uBViHcYLOb2280tEpr64iB9b6YRkWil3EODiiLd9JS3V7+BWpZVF8mbGy8AUR7T3GUsYmEvhHGw2s0IosOUVIu5W3eiU/K4CFMEsmV5JZLRS...Yeb/PX6yl+4vG17kFDiBkCPl+88lcbS5iHc1mXuWePlW+Q/6tN55Oy6mX8J/dGLVBYBQKbmChGuRlZdwWJgHh+O5rN5aDAEda/aYqfgdUC8BHSwhCQPtyCIsOKjOYZzM`. So the being and end line as well as the `=JfHr`. I will add something to the doc for this.

Reply:
Nice! I got curious and this is the bash one liner I got:

```
gpg --export-secret-keys --armor $KEY_ID |grep -v '\-\-' |grep -v '=.' |tr -d '\n'
```

Reply:
you can also do multiline with trailing `\` chars

Reply:
> Nice! I got curious and this is the bash one liner I got:
> 
> ```
> gpg --export-secret-keys --armor $KEY_ID |grep -v '\-\-' |grep -v '=.' |tr -d '\n'
> ```

This didn't work for me because I had an `=` in the last line. I came up with the following that trims the top and bottom lines and removes new line characters

`gpg --export-secret-keys --armor $KEY_ID |tac |tail -n +3 |tac |tail -n +3 |tr -d '\n'`